activerecord/lib/active_record/connection_adapters/abstract/schema_definitions.rb

```diff
@@ -539,7 +539,7 @@ def new_column_definition(name, type, **options) # :nodoc:
         type = aliased_types(type.to_s, type)
 
         if @conn.supports_datetime_with_precision?
-          if type == :datetime && !options.key?(:precision)
+          if type == :datetime && (!options.key?(:precision) || options[:precision].nil?)
```

Comment:
I'm not sure we want this behavior. If I'm not mistaken, the expected behavior is

<table><tr><td>

```ruby
t.datetime :name
```

</td><td>

Implicitly use Active Record's default precision:

- `precision: 6` as of Rails 7
- `precision: nil` prior to that

</td></tr><tr><td>

```ruby
t.datetime :name, precision: 3
```

</td><td>

Explicitly use three decimals of precision.

</td></tr><tr><td>

```ruby
t.datetime :name, precision: 0
```

</td><td>

Explicitly use zero decimals of precision.

</td></tr><tr><td>

```ruby
t.datetime :name, precision: nil
```

</td><td>

Explicitly use whatever the database's default precision is.

</td></tr><tr><table>